### PR TITLE
Changes references from pg 9.5 to 9.6

### LINKED ISSUE #191: Update usage of PostgreSQL 9.5 in docs with 9.6
```
grep -R -l "9.5" *
aboutcitus/introduction_to_citus.rst
admin_guide/cluster_management.rst
admin_guide/upgrading_citus.rst
cloud/logging.rst
dist_tables/ddl.rst
dist_tables/dml.rst
dist_tables/extensions.rst
dist_tables/querying.rst
images/aws_cluster_launch.png
images/aws_parameters.png
images/citus-basic-arch.png
images/citus_logs_add_log_form.png
images/citus_logs_dashboard.png
images/cloud_logs_papertrail.png
images/cloud_provisioning.png
installation/multi_machine_aws.rst
installation/production_deb.rst
installation/production_rhel.rst
installation/requirements.rst
installation/single_machine_deb.rst
installation/single_machine_osx.rst
installation/single_machine_rhel.rst
performance/performance_tuning.rst
performance/query_processing.rst
reference/append.rst
reference/citus_sql_reference.rst
reference/configuration.rst
reference/metadata_tables.rst
reference/user_defined_functions.rst
Sumedhs-MBP:citus_docs Pathak$ 
```

### PR DESCRIPTION
**Pull request by begriffs**
Fixes #191

Changed all occurrences except in `installation/requirements.rst` where it says, "Citus 5.2 requires PostgreSQL 9.5 or later versions."

Also spidered the docs checking for broken links. Did not find any from the version renaming but did find some for the mobile nav menu. Included a fix in this PR as well.

### CODE REVIEW

**Comment by begriffs** [comment]
The installation instructions refer to https://install.citusdata.com/community/rpm.sh which is still on postgres 9.5. We'll need an updated version of this for 9.6, perhaps renamed rpm-96.sh so that the old docs remain valid.

File: _themes/citus/layout.html [hunk 1]
```diff
@@ -164,9 +164,10 @@
       <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
         <a href="{{ pathto(master_doc) }}">{{ project }}</a>
       <ul>
-        <li><a href="/product/citus">Product</a></li>
-        <li><a href="/solutions/applications">Solutions</a></li>
-        <li><a href="/blog">Blog</a></li>
+        <li><a href="https://www.citusdata.com/product/citus">Product</a></li>
```
Comment (begriffs) [inline _themes/citus/layout.html]: After changing version numbers in a bunch of postgres docs links I ran a program checking for broken links in all these changes. Did not find any from the version renaming but did find broken links in the mobile nav menu. I Included the fix in this PR as well.

File: installation/single_machine_rhel.rst [hunk 2]
```diff
@@ -7,7 +7,7 @@ Fedora, CentOS, or Red Hat
 
 This section describes the steps needed to set up a single-node Citus cluster on your own Linux machine from RPM packages.
 
-**1. Install PostgreSQL 9.5 and the Citus extension**
+**1. Install PostgreSQL 9.6 and the Citus extension**
```
Comment (sumedhpathak) [inline installation/single_machine_rhel.rst]: Update citus_95 to citus_96

File: installation/production_deb.rst [hunk 3]
```diff
@@ -24,24 +24,24 @@ Steps to be executed on all nodes
...
   # install the server and initialize db
-  sudo apt-get -y install postgresql-9.5-citus
+  sudo apt-get -y install postgresql-9.6-citus
 
   # preload citus extension
-  sudo pg_conftool 9.5 main set shared_preload_libraries citus
+  sudo pg_conftool 9.6 main set shared_preload_libraries citus
 
-This installs centralized configuration in `/etc/postgresql/9.5/main`, and creates a database in `/var/lib/postgresql/9.5/main`.
+This installs centralized configuration in `/etc/postgresql/9.6/main`, and creates a database in `/var/lib/postgresql/9.6/main`.
```
Comment (begriffs) [inline installation/production_deb.rst]: Verified that these paths are accurate on Ubuntu 16.04 when I install Postgres 9.6 without Citus. The particular paths differ on RHEL and I did not verify them.

**Comment by begriffs** [comment]
I think it may be more productive to fully test install instructions once the new Citus is released. Otherwise testing is more difficult and also not completely representative of the process.

**Comment by sumedhpathak** [comment]
Pushed a few more updates to this PR.

**Comment by sumedhpathak** [comment]
Approving changes, after pushing a few more fixes. Before we push, we'll want to test the instructions thoroughly with the packages to validate these changes. Let's do that tomorrow.

**Pull request merged**